poly-commit/src/linear_codes/mod.rs

```diff
@@ -416,25 +409,16 @@ where
     where
         Self::Commitment: 'a,
     {
-        let labeled_commitments: Vec<&'a LabeledCommitment<Self::Commitment>> =
-            commitments.into_iter().collect();
-        let values: Vec<F> = values.into_iter().collect();
-
-        if labeled_commitments.len() != proof_array.len()
-            || labeled_commitments.len() != values.len()
-        {
-            return Err(Error::IncorrectInputLength(
-                format!(
-                    "Mismatched lengths: {} proofs were provided for {} commitments with {} claimed values",labeled_commitments.len(), proof_array.len(), values.len()
-                )
-            ));
-        }
+        let mut values = values.into_iter();
+
         let leaf_hash_param: &<<C as Config>::LeafHash as CRHScheme>::Parameters =
             vk.leaf_hash_param();
         let two_to_one_hash_param: &<<C as Config>::TwoToOneHash as TwoToOneCRHScheme>::Parameters =
             vk.two_to_one_hash_param();
 
-        for (i, labeled_commitment) in labeled_commitments.iter().enumerate() {
+        let mut i = 0;
+        for labeled_commitment in commitments {
```

Comment:
Again here, maybe we use `enumerate` or `zip` the `proof_array` to `commitments`. Carrying the index looks a bit ugly to me. :D